openbb_platform/providers/fmp/openbb_fmp/models/financial_ratios.py

```diff
@@ -226,37 +237,57 @@ async def aextract_data(
 
         base_url = "https://financialmodelingprep.com/api/v3"
 
-        ttm_url = f"{base_url}/ratios-ttm/{query.symbol}?&apikey={api_key}"
+        ttm_dict = {"period": "TTM", "date": datetime.now().strftime("%Y-%m-%d")}
 
-        url = (
-            f"{base_url}/ratios/{query.symbol}?"
-            f"period={query.period}&limit={query.limit}&apikey={api_key}"
-            if query.period != "ttm"
-            else ttm_url
-        )
-        results = await amake_request(
-            url, response_callback=response_callback, **kwargs
-        )
+        include_ttm = query.period != "ttm" and query.with_ttm
+
+        async def response_callback(
+            response: ClientResponse, session: ClientSession
+        ) -> List[Dict]:
+            results: List[dict] = await response.json()  # type: ignore
```

Comment:
This effectively undoes the efforts to expose the actual error message from the FMP response - https://github.com/OpenBB-finance/OpenBBTerminal/pull/6237. 

Instead of clearly communicating the issue to the user:
```
OpenBBError: FMP Error Message -> Invalid API KEY. Feel free to create a Free API Key or visit https://site.financialmodelingprep.com/faqs?search=why-is-my-api-key-invalid for more information.
```

It is now a validation error that cannot be traced back to the root cause:

```
OpenBBError: 2 validation errors for FMPFinancialRatiosData
periodEnding
  Field required [type=missing, input_value={'error_message': 'Invali... for more information.'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
fiscalPeriod
  Field required [type=missing, input_value={'error_message': 'Invali... for more information.'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/missing
```